BrightID/src/components/NewConnectionsScreens/MyCodeScreen.js

```diff
@@ -29,103 +28,65 @@ import { startConnecting, stopConnecting } from './actions/connecting';
  *
  */
 
-type State = {
-  copied: boolean,
-  countdown: number,
-  qrsvg:
-    | string
-    | {
-        svg: {
-          $: {
-            viewBox: string,
-          },
-        },
-      },
-};
-
 const COPIED_TIMEOUT = 500;
 
-export class MyCodeScreen extends React.Component<Props, State> {
-  timer: IntervalID;
+const Container = DEVICE_IOS ? SafeAreaView : View;
 
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      qrsvg: '',
-      copied: false,
-      countdown: 0,
-    };
-  }
+export const MyCodeScreen = (props) => {
+  const dispatch = useDispatch();
+  const name = useSelector((state) => state.user.name);
+  const ttl = useSelector((state) => state.connectQrData.myQrData?.ttl ?? 0);
+  const qrString = useSelector(
+    (state) => state.connectQrData.myQrData?.qrString,
+  );
+  const timestamp = useSelector(
+    (state) => state.connectQrData.myQrData?.timestamp ?? Date.now(),
+  );
 
-  componentDidMount() {
-    // we need to reload the QR code every mount
-    const { dispatch } = this.props;
-    dispatch(stopConnecting());
-    dispatch(startConnecting());
-    // start local timer to display countdown
-    this.timer = setInterval(() => {
-      this.timerTick();
-    }, 100);
+  const [qrsvg, setQrsvg] = useState('');
+  const [copied, setCopied] = useState(false);
 
-    // For now directly jump to preview when a profile is received.
-    // Needs to be replaced by notification system.
-    // Or maybe leave it like that, so as long as the MyCode screen is open the first responder
-    // directly to the preview contact. Any subsequent responders will end up in the
-    // notification area.
-    emitter.on('connectDataReady', this.navigateToPreview);
-  }
+  const [countdown, setCountdown] = useState(ttl - (Date.now() - timestamp));
+  console.log('countdown', countdown);
 
-  componentDidUpdate(prevProps: Props, prevState: State) {
-    const currentQrString = this.props.myQrData
-      ? this.props.myQrData.qrString
-      : undefined;
-    const prevQrString = prevProps.myQrData
-      ? prevProps.myQrData.qrString
-      : undefined;
-    if (currentQrString !== prevQrString) {
-      this.checkQrCode();
+  const timerTick = () => {
+    let countdown = ttl - (Date.now() - timestamp);
+    if (countdown <= 0) {
+      dispatch(startConnecting());
```

Comment:
I think this is not necessary, as checkQrCode() whithin the useEffect block below will already dispatch startConnecting() as soon as the qrString from redux is not set.